### DISCUSSION TITLE
quarto

### DISCUSSION THREAD

**Discussion by Polkas** [Polkas/shinyreporter#3] [Ideas]
I think we should use quarto for renderer engine, https://quarto.org/

**Comment by kpagacz** [comment]
There are now discussion on github! Holy.

There are options, certainly. Very similar to just `jupyter` renderers.

The big problem with `quarto` and anything that is not native python is that we will have to ship it with the package. So we will have to introduce a dependency on either:
* a binary shipped with the package
* some javascript bundled with the application + JS runtime
* something else completely

From what I can tell `quarto` is written in JS + Typescript. Underneath all that, they seem to be using `pandoc` https://github.com/search?q=repo%3Aquarto-dev%2Fquarto%20pandoc&type=code

I don't think I have a good solution to that right now. Ideally we can just ship a bunch of python code and not worry about dependencies... :(

**Reply by kpagacz** [reply]
But I don't think we will manage that

**Comment by Polkas** [comment]
You might be right that we should use jupyter an nbconvert as they are native solutions.
https://github.com/jupyter/nbconvert

**Reply by Polkas** [reply]
jupyter can be converted eaisly to:
HTML
LaTeX
PDF
Reveal JS
Markdown (md)
ReStructured Text (rst)
executable script

**Reply by Polkas** [reply]
quarto can only be recommended as one of the ways to convert a jupyter

**Reply by Polkas** [reply]
https://stackoverflow.com/a/45672031/5442527 nbformat can be used to build jupyter by script calls

**Comment by kpagacz** [comment]
Then, I guess we have a tentative consensus. We build an `nbformat` file and worry about rendering that to anything else later. 

BTW, from what I can see nbconvert is not native either. https://nbconvert.readthedocs.io/en/latest/install.html It uses pandoc, tex and playwright for rendering to html/markdown, pdf and webpdf.
